include/deal.II/lac/vector.templates.h

```diff
@@ -120,23 +120,38 @@ namespace internal
 
   template <typename Functor>
   void vectorized_transform(Functor &functor,
-                            size_type vec_size)
+                            size_type vec_size,
+                            parallel::internal::TBBPartitioner *partitioner = NULL)
   {
 #ifdef DEAL_II_WITH_THREADS
     // only go to the parallel function in case there are at least 4 parallel
     // items, otherwise the overhead is too large
     if (vec_size >= 4*internal::Vector::minimum_parallel_grain_size &&
         MultithreadInfo::n_threads() > 1)
       {
+        // in case the partitioner was used by another thread, create a
+        // private partitioner here for use in the loop (we could also use the
+        // auto partitioner of TBB but that increases code size considerably
+        // without major benefit)
+        tbb::affinity_partitioner *tbb_partitioner = (partitioner == NULL) ?
+                                                     NULL : partitioner->acquire();
+        tbb::affinity_partitioner private_partitioner;
```

Comment:
No, they are very cheap, the just set the length of an array to zero and the pointer to zero. But I agree that it is neater to let the class do the cleanup work. I will use a solution similar to what you propose.